src/sortStudents.ts

```
@@ -1,16 +1,68 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'Name',
+  Surname = 'Surname',
+  Age = 'Age',
+  Married = 'Married',
+  AverageGrade = 'AverageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((a, b) => {
+    let valueA: number | string;
+    let valueB: number | string;
+
+    switch (sortBy) {
+      case SortType.Name:
+        valueA = a.name;
+        valueB = b.name;
+        break;
+      case SortType.Surname:
+        valueA = a.surname;
+        valueB = b.surname;
+        break;
+      case SortType.Age:
+        valueA = a.age;
+        valueB = b.age;
+        break;
+      case SortType.Married:
+        valueA = Number(a.married);
+        valueB = Number(b.married);
+        break;
+      case SortType.AverageGrade:
+        valueA = a.grades.reduce((total, grade) => total
+          + grade, 0) / a.grades.length;
+
+        valueB = b.grades.reduce((total, grade) => total
+          + grade, 0) / b.grades.length;
+        break;
+      default:
+        return 0;
```

Returning 0 in the default case of the switch statement is fine, but it would be better to handle this case explicitly or throw an error, as it indicates an unexpected `sortBy` value.